client/src/panels/panel_declarations/misc/key_concept_panels/KeyConcepts.js

```diff
@@ -0,0 +1,160 @@
+import classNames from "classnames";
+import { ButtonToolbar } from "react-bootstrap";
+import MediaQuery from "react-responsive";
+import { withRouter } from "react-router";
+
+import { has_local_storage } from "src/core/feature_detection.js";
+
+import { util_components, breakpoints } from "../../shared.js";
+
+import "./KeyConcepts.scss";
+
+import common_lang from "./common_questions.yaml";
+import fin_lang from "./financial_questions.yaml";
+import ppl_lang from "./people_questions.yaml";
+import results_lang from "./results_questions.yaml";
+import tag_lang from "./tagging_questions.yaml";
+
+const {
+  create_text_maker_component,
+  AutoAccordion,
+  KeyConceptList,
+} = util_components;
+
+const { text_maker, TM } = create_text_maker_component([
+  common_lang,
+  fin_lang,
+  ppl_lang,
+  results_lang,
+  tag_lang,
+]);
+
+class KeyConcepts_ extends React.Component {
+  state = {
+    sticky: false,
+    should_pin: false,
+    key_concepts_width: null,
+  };
+  constructor(props) {
+    super(props);
+    this.accordionRef = React.createRef(null);
+
+    let should_pin;
+    if (has_local_storage) {
+      try {
+        should_pin = JSON.parse(
+          localStorage.getItem(`should_pin_key_concepts`)
+        );
+      } catch {
+        should_pin = null;
+      }
+    }
+
+    this.keyConceptsContainerRef = React.createRef();
+
+    this.state = {
```

Comment:
two initial state declaration, I think better to keep it at one place

Reply:
Also I'm not sure why there are two different states. What different roles are they serving? What does `sticky` mean, and what does `should_pin` mean? Can you also come up with better names? Something like `is_pinned` that name show the state more intuitively?

Reply:
whoops! I forgot to remove the state outside of the constructor after moving it into there. There's only one state in reality even if it's declared twice though! (I'm pretty sure at least). I've renamed the variables/keys to more appropriately match their uses.